src/pubsub.ts

```diff
@@ -889,7 +889,7 @@ export class PubSub {
   getClient_(config: GetClientConfig, callback: GetClientCallback) {
     const hasProjectId =
       this.projectId && this.projectId !== PROJECT_ID_PLACEHOLDER;
-    if (!hasProjectId && !this.isEmulator) {
+    if (!hasProjectId) {
```

Comment:
I wonder if the check `!this.isEmulator` should still be left in, because it is valid that this should pull from auth client when it is not in emulator.

Now when it is emulator mode, is it valid to pull from auth? Right below line 903, i did something like so:
```js
        if (!hasProjectId && this.isEmulator) {
            const {GOOGLE_CLOUD_PROJECT, PUBSUB_PROJECT_ID} = process.env;
            const projectId = PUBSUB_PROJECT_ID || GOOGLE_CLOUD_PROJECT || 'pubsub-emulator';
            this.projectId = projectId;
            this.options.projectId = projectId;
        }
```

and it seems to work out fine (user can set PUBSUB_PROJECT_ID || GOOGLE_CLOUD_PROJECT or neither, because it falls back to a project id called 'pubsub-emulator' that I made up to ensure that the subscriber and the publisher are in the same project.)

However, this raises the question for the documentation from https://cloud.google.com/pubsub/docs/emulator#automatically_setting_the_variables:

> You can optionally set the PUBSUB_PROJECT_ID environment variable for the project you want to use for the emulator, but you only need to set PUBSUB_EMULATOR_HOST for your application to connect to the emulator

In my proposal, if I changed

```js
const projectId = PUBSUB_PROJECT_ID || GOOGLE_CLOUD_PROJECT || 'pubsub-emulator';
```

to 

```js
const projectId = PUBSUB_PROJECT_ID || GOOGLE_CLOUD_PROJECT;
```

the topic and subscription can't be found despite it being created, which I didn't understand - so there's a discrepancy between this package and the documentation itself.

Reply:
We actually went down this route as a feature request from users wanting to use the project ID returned from the auth client. I've made a commit that still requests the project ID from the auth client but ignores any errors in the event that the user is using the emulator.